### PR TITLE
Support hcd

### LINKED ISSUE #1320: Add support for HCD
HCD is reaching the preview stage and we'd need to support it in the operator.

### PR DESCRIPTION
**Pull request by adejanovski**
**What this PR does**:
Adds support for HCD as server type and remove the regex checks for server versions which already exist in cass-operator.

Fixes #1320 

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: apis/k8ssandra/v1alpha1/k8ssandracluster_types.go [hunk 1]
```diff
@@ -530,12 +530,24 @@ type ServerDistribution string
...
+func (sd *ServerDistribution) IsDse() bool {
+	return *sd == ServerDistributionDse
+}
+
 func (kc *K8ssandraCluster) DefaultNumTokens(serverVersion *semver.Version) float64 {
-	if kc.Spec.Cassandra.ServerType == ServerDistributionCassandra && serverVersion.Major() == 3 {
+	if kc.Spec.Cassandra.ServerType.IsCassandra() && serverVersion.Major() == 3 {
 		return float64(256)
 	}
+	if kc.Spec.Cassandra.ServerType.IsCassandra() {
```
Comment (burmanm) [inline apis/k8ssandra/v1alpha1/k8ssandracluster_types.go:548]: Isn't this breaking backwards compatibility? 16 is the default for OSS versions and all previous deployments would use that.

File: charts/k8ssandra-operator/crds/k8ssandra-operator-crds.yaml [hunk 2]
```diff
@@ -10258,7 +10258,7 @@ spec:
                             ServerVersion is the Cassandra or DSE version. The following versions are supported:
                             - Cassandra: 3.11.X, 4.X.X and 5.X.X
                             - DSE: 6.8.X, 7.x.x
-                          pattern: (6\.8\.\d+)|(3\.11\.\d+)|(4\.\d+\.\d+)|(5\.\d+\.\d+)|(7\.\d+\.\d+)
+                            - HCD: 1.0.x
```
Comment (burmanm) [inline charts/k8ssandra-operator/crds/k8ssandra-operator-crds.yaml:10261]: suggestion: We shouldn't remove pattern entirely, it should be at least something like ``\d+\.\d+\.\d+``. We're not supporting versions like ``6.9`` that would auto-update to newest point-release for example.

File: apis/k8ssandra/v1alpha1/k8ssandracluster_types.go (hunk 1 above)
Comment (adejanovski) [reply apis/k8ssandra/v1alpha1/k8ssandracluster_types.go:548]: you're right, and I shouldn't have touched this anyway here.

File: charts/k8ssandra-operator/crds/k8ssandra-operator-crds.yaml (hunk 2 above)
Comment (adejanovski) [reply charts/k8ssandra-operator/crds/k8ssandra-operator-crds.yaml:10261]: fair enough 👍

**Comment by adejanovski** [comment]
@burmanm , I addressed your comments. Could you review again?

**Review by burmanm — APPROVED** [review]

**Pull request merged**